src/main/java/ch/naviqore/raptor/impl/RaptorConnection.java

```diff
@@ -15,7 +15,7 @@
 @NoArgsConstructor(access = AccessLevel.PACKAGE)
 @Getter
 @ToString
-class RaptorConnection implements Connection {
+class RaptorConnection implements Connection, Comparable<Connection> {
```

Comment:
Why implement the Comparable? I still don't see a reason to keep this.

Reply:
This is exactly the problem that I try to explain, we don't use it, but we should.

We should sort all our returns in `RaptorAlgorithm / Router`, by calling sort before return in the `LabelProcessor`:

```
return connections.stream().sorted().toList();
```
Which will then implicitly call the `compareTo` required by the `Comparable` Interface.

If we do this now, most of our tests fail. This is not because the Raptor is not working, but simply due to sorting issues. The problem is that the way we implement the Raptor currently influences the order of the results, which is not very stable. Additionally, if we implement further versions (native or mcRaptor), this instability will cause further issues.